os-server/src/main/java/app/openschool/course/discussion/mentor/question/MentorQuestionServiceImpl.java

```diff
@@ -48,6 +49,21 @@ public QuestionResponseDto create(QuestionRequestDto requestDto, String email) {
         mentorQuestionRepository.save(creteQuestion(requestDto, email)));
   }
 
+  @Override
+  public QuestionResponseDto update(Long id, String text) {
+    MentorQuestion mentorQuestion =
+        mentorQuestionRepository.findById(id).orElseThrow(IllegalArgumentException::new);
```

Comment:
instead of find by id you can select by user id and question id so you will not need to do a check one layer above